lib/core/macros.nim

```diff
@@ -883,28 +885,32 @@ proc treeTraverse(n: NimNode; res: var string; level = 0; isLisp = false, indent
     res.add(" " & $n.strVal.newLit.repr)
   of nnkNone:
     assert false
+  elif n.kind in {nnkOpenSymChoice, nnkClosedSymChoice} and collapseSymChoice:
+    res.add(" " & $n.len)
+    if n.len > 0:
+      res.add(" " & $n[0].strVal.newLit.repr)
```

Comment:
This assumes that at position 0 there is a node for which `.strVal` is valid and afaict you can construct trees where this is false. The `Repr` procs must be able to turn any tree into a string representation even if that tree is invalid.

Reply:
fixed